src/components/new-safe/steps/Step0/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { Box, Button, Divider, Grid, Paper, Typography } from '@mui/material'
+import useWallet from '@/hooks/wallets/useWallet'
+import ChainSwitcher from '@/components/common/ChainSwitcher'
+import useOnboard, { type ConnectedWallet, connectWallet } from '@/hooks/wallets/useOnboard'
+import NetworkSelector from '@/components/common/NetworkSelector'
+import useIsWrongChain from '@/hooks/useIsWrongChain'
+import { useCurrentChain } from '@/hooks/useChains'
+import { isPairingSupported } from '@/services/pairing/utils'
+
+import type { NewSafeFormData } from '@/components/new-safe/CreateSafe'
+import type { StepRenderProps } from '@/components/new-safe/CardStepper/useCardStepper'
+import KeyholeIcon from '@/components/common/icons/KeyholeIcon'
+import { OVERVIEW_EVENTS, trackEvent } from '@/services/analytics'
+import PairingQRCode from '@/components/common/PairingDetails/PairingQRCode'
+import PairingDescription from '@/components/common/PairingDetails/PairingDescription'
+
+export const ConnectWalletContent = ({
+  wallet,
+  isWrongChain,
+  setStep,
+}: {
+  wallet: ConnectedWallet | null
+  isWrongChain: boolean
+  setStep: StepRenderProps<NewSafeFormData>['setStep']
+}) => {
+  const chain = useCurrentChain()
+  const isSupported = isPairingSupported(chain?.disabledWallets)
+  const onboard = useOnboard()
+
+  const handleConnect = async () => {
+    if (!onboard) return
+
+    trackEvent(OVERVIEW_EVENTS.OPEN_ONBOARD)
+
+    await connectWallet(onboard)
+    setStep(1)
```

Comment:
Probably we should call do something like 
```ts
onSubmit({ owners: [{ address: wallet.address }]
```
instead such that the owners step will have the connected wallet as owner. 
Otherwise it will not be prefilled with the connected wallet as the initialData of the stepper was already created.

Reply:
Good idea! I reused the `WalletDetails` component which allows us to pass an `onConnect` handler maybe we can solve it that way too.